src/TesApi.Web/BatchPool.cs

```diff
@@ -481,19 +481,21 @@ public async ValueTask CreatePoolAndJobAsync(Microsoft.Azure.Management.Batch.Mo
             try
             {
                 CloudPool pool = default;
-                var poolInfo = await _azureProxy.CreateBatchPoolAsync(poolModel, isPreemptible);
-
                 await Task.WhenAll(
-                    Task.Run(async () => pool = await _azureProxy.GetBatchPoolAsync(poolInfo.PoolId, new ODATADetailLevel { SelectClause = CloudPoolSelectClause }, cancellationToken)),
-                    _azureProxy.CreateBatchJobAsync(poolInfo, cancellationToken));
+                    _azureProxy.CreateBatchJobAsync(new() { PoolId = poolModel.Name }, cancellationToken),
+                    Task.Run(async () =>
+                    {
+                        var poolInfo = await _azureProxy.CreateBatchPoolAsync(poolModel, isPreemptible);
+                        pool = await _azureProxy.GetBatchPoolAsync(poolInfo.PoolId, new ODATADetailLevel { SelectClause = CloudPoolSelectClause }, cancellationToken);
+                    }, cancellationToken));
 
                 Configure(pool);
                 _ = _batchPools.AddPool(this);
             }
             catch (AggregateException ex)
             {
                 var exception = ex.Flatten();
-                if (exception.InnerExceptions.Count < 2)
+                if (exception.InnerExceptions?.Count != 1)
```

Comment:
can you add a comment?  It seems oddly specific.  Only do something if 0 or >1 InnerExceptions ?